### PR TITLE
[plugin.video.zdf_de_2016] 1.0.0

### PR DESCRIPTION
**Pull request by generia**
##### Kodi Addon "ZDF Mediathek 2016"
ZDF is a German public television channel providing it's television shows online in their so called "ZDF Mediathek".

This is yet another version of the "ZDF Mediathek", since the existing addons stopped working due to the website relaunch end of October 2016 and updates of the existing plugins were not yet available my wife forced me to take action.

*Features*
- combined search function over multiple result pages
- optimized for fast HTML parsing of the zdf.de website

*Compatibility*
This addon was tested with 
- Kodi 16.1 "Jarvis", Confluence Skin, MacOS
- Kode 17.0 "Krypton" beta 5, Estouchy Skin, MacOS/Linux 

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by generia** [comment]
see also my comments in #758

File: plugin.video.zdf_de_2016/addon.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        if item is None:
+            return
+
+        title = item.title
+
+        infoLabels = {}
+        infoLabels['title'] = title
+        infoLabels['sorttitle'] = title
+        infoLabels['genre'] = item.genre
+        infoLabels['plot'] = item.text
```
Comment (phil65) [inline plugin.video.zdf_de_2016/addon.py]: please also set the correct mediatype (probably "video" in this case.

File: plugin.video.zdf_de_2016/addon.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.zdf_de_2016" name="ZDF Mediathek 2016" version="1.0.0" provider-name="Generia">
+  <requires>
+    <import addon="xbmc.python" version="2.1.0"/>
```
Comment (phil65) [inline plugin.video.zdf_de_2016/addon.xml]: this should be 2.24.0

File: plugin.video.zdf_de_2016/de/generia/kodi/plugin/frontend/zdf/SearchPage.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        self._progress = xbmcgui.DialogProgress()
+        try:
+            msg = self._(32021)
+            if pages != -1:
+                msg = self._(32022, page, pages)
+            self._progress.create(self._(32020), msg)
+            self._progress.update(0, msg)
+            self._loadResults(request, response, apiToken, pages, page, query)
+        except:
+            self.warn("Timer - loading results ... expeption")
```
Comment (phil65) [inline plugin.video.zdf_de_2016/de/generia/kodi/plugin/frontend/zdf/SearchPage.py]: expeption? :)

File: plugin.video.zdf_de_2016/de/generia/kodi/plugin/backend/zdf/Teaser.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+labelPattern = getTagPattern('div', 'teaser-label')
+iconPattern = compile('<span\s*class="icon-[0-9]*_([^ ]*) icon">')
+catPattern = compile('<span class="teaser-cat"[^>]*>([^<]*)</span>')
+aPattern = compile('<a href="([^"]*)"[^>]*>')
+titleIconPattern = compile('<span\s*class="title-icon icon-[0-9]*_([^"]*)">')
+textPattern = compile('<p class="teaser-text"[^>]*>([^<]*)</p>')
+datePattern = compile('<dd class="video-airing"[^>]*>([^<]*)</dd>')
+
+    
+def compareTeasers(t1, t2):
```
Comment (phil65) [inline plugin.video.zdf_de_2016/de/generia/kodi/plugin/backend/zdf/Teaser.py]: you could also just do "int(t2 is None) - int (t2 is None)"  :)

File: plugin.video.zdf_de_2016/de/generia/kodi/plugin/frontend/zdf/PlayVideo.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                self._parse(streamInfo)
+                
+                url = streamInfo.streamUrl
+                if url is None:
+                    self.warn("PlayVideo - can't find stream-url in stream-info-url '{1}' in content '{2}'", videoContent.streamInfoUrl, streamInfo.content)
+                    response.sendError(self._(32012) + " '" + contentName +"'", Action('SearchPage'))
+                    return
+
+                image = videoContent.image
+                item = xbmcgui.ListItem(videoContent.title)
```
Comment (phil65) [inline plugin.video.zdf_de_2016/de/generia/kodi/plugin/frontend/zdf/PlayVideo.py]: also here, please set mediatype.

**Comment by phil65** [comment]
Looks good in general. Thx and sorry for the delays.

**Comment by generia** [comment]
Hi Phil,
thank for your review. I adopted your remarks in the next version.
It contains additionally a number of bug-fixes and improvements.
I created a new pull-request #783 in order to trigger the coday analysis, but for some reason this did not start.
The commit in the new pull-request #783 contains all files for this addon, i.e. the files from this pull-request and the new/changed ones.
So I would ask you to review #783 and we can close this request then.

Thanks and best regards.

**Pull request closed without merging**